src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificateRevocationListBuilder.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Formats.Asn1;
+using System.Numerics;
+using System.Security.Cryptography.Asn1;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    public sealed partial class CertificateRevocationListBuilder
+    {
+        private readonly List<RevokedCertificate> _revoked;
+        private AsnWriter? _writer;
+
+        public CertificateRevocationListBuilder()
+        {
+            _revoked = new List<RevokedCertificate>();
+        }
+
+        private CertificateRevocationListBuilder(List<RevokedCertificate> revoked)
+        {
+            Debug.Assert(revoked != null);
+            _revoked = revoked;
+        }
+
+        public void AddEntry(X509Certificate2 certificate)
+        {
+            AddEntry(certificate, DateTimeOffset.UtcNow);
+        }
+
+        public void AddEntry(X509Certificate2 certificate, DateTimeOffset revocationTime)
+        {
+            ArgumentNullException.ThrowIfNull(certificate);
+
+            AddEntry(certificate.SerialNumberBytes.Span, revocationTime);
+        }
+
+        public void AddEntry(byte[] serialNumber)
+        {
+            AddEntry(serialNumber, DateTimeOffset.UtcNow);
+        }
+
+        public void AddEntry(byte[] serialNumber, DateTimeOffset revocationTime)
+        {
+            ArgumentNullException.ThrowIfNull(serialNumber);
+
+            AddEntry(new ReadOnlySpan<byte>(serialNumber), revocationTime);
+        }
+
+        public void AddEntry(ReadOnlySpan<byte> serialNumber)
+        {
+            AddEntry(serialNumber, DateTimeOffset.UtcNow);
+        }
+
+        public void AddEntry(ReadOnlySpan<byte> serialNumber, DateTimeOffset revocationTime)
+        {
+            if (serialNumber.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptyOrNullArray, nameof(serialNumber));
+
+            _revoked.Add(
+                new RevokedCertificate
+                {
+                    Serial = serialNumber.ToArray(),
+                    RevocationTime = revocationTime.ToUniversalTime(),
+                });
+        }
+
+        public void ExpireEntries(DateTimeOffset oldestRevocationTimeToKeep)
+        {
+            _revoked.RemoveAll(rc => rc.RevocationTime < oldestRevocationTimeToKeep);
+        }
+
+        public void RemoveEntry(byte[] serialNumber)
+        {
+            ArgumentNullException.ThrowIfNull(serialNumber);
+
+            RemoveEntry(new ReadOnlySpan<byte>(serialNumber));
+        }
+
+        public void RemoveEntry(ReadOnlySpan<byte> serialNumber)
```

Comment:
Now that I look at it, what do you think about returning a `bool` here to indicate if it was removed or not? I think many other collection type's removal do something like that.

Reply:
Yeah, that seems fine.